rune/libenclave/internal/runtime/pal/skeleton/Makefile

```diff
@@ -9,45 +9,33 @@ ENCL_CFLAGS := -Wall -Werror -static -nostdlib -nostartfiles -fPIC \
 	       -fno-stack-protector -mrdrnd -std=gnu11
 HOST_LDFLAGS := -fPIC -shared -Wl,-Bsymbolic
 
-IS_SGX_FLC ?= $(shell if lscpu | grep -q sgx_lc; then echo "1"; else echo "0"; fi;)
-
 PRODUCT_ENCLAVE ?=
 
-TEST_CUSTOM_PROGS := $(OUTPUT)/encl.bin $(OUTPUT)/encl.ss $(OUTPUT)/liberpal-skeleton-v1.so $(OUTPUT)/liberpal-skeleton-v2.so $(OUTPUT)/liberpal-skeleton-v3.so $(OUTPUT)/signing_key.pem
-
-ifeq ($(IS_SGX_FLC),0)
-        TEST_CUSTOM_PROGS += $(OUTPUT)/encl.token
-endif
+TEST_CUSTOM_PROGS := $(OUTPUT)/encl.bin $(OUTPUT)/encl.ss $(OUTPUT)/aesm.pb-c.c $(OUTPUT)/liberpal-skeleton-v1.so $(OUTPUT)/liberpal-skeleton-v2.so $(OUTPUT)/liberpal-skeleton-v3.so $(OUTPUT)/signing_key.pem
 
 all: $(TEST_CUSTOM_PROGS)
 
-$(OUTPUT)/liberpal-skeleton-v1.so: $(OUTPUT)/sgx_call.o $(OUTPUT)/liberpal-skeleton-v1.o $(OUTPUT)/liberpal-skeleton.o $(OUTPUT)/sgxutils.o
-	$(CC) $(HOST_LDFLAGS) -o $@ $^
+SOURCES = $(filter-out encl.c sgxsign.c, $(wildcard *.c)) aesm.pb-c.c
+OBJS = $(patsubst %.c, %.o, $(SOURCES))
 
-$(OUTPUT)/liberpal-skeleton-v1.o: liberpal-skeleton-v1.c liberpal-skeleton.c
-	$(CC) $(HOST_CFLAGS) -c $< -o $@
+$(OUTPUT)/liberpal-skeleton-v1.so: $(OUTPUT)/sgx_call.o $(OUTPUT)/liberpal-skeleton-v1.o $(OUTPUT)/liberpal-skeleton.o $(OUTPUT)/sgxutils.o $(OUTPUT)/aesm.o $(OUTPUT)/aesm.pb-c.o
+	$(CC) $(HOST_LDFLAGS) -o $@ $^ -lprotobuf-c
 
-$(OUTPUT)/liberpal-skeleton-v2.so: $(OUTPUT)/sgx_call.o $(OUTPUT)/liberpal-skeleton-v2.o $(OUTPUT)/liberpal-skeleton.o $(OUTPUT)/sgxutils.o
-	$(CC) $(HOST_LDFLAGS) -o $@ $^
+$(OUTPUT)/liberpal-skeleton-v2.so: $(OUTPUT)/sgx_call.o $(OUTPUT)/liberpal-skeleton-v2.o $(OUTPUT)/liberpal-skeleton.o $(OUTPUT)/sgxutils.o $(OUTPUT)/aesm.o $(OUTPUT)/aesm.pb-c.o
+	$(CC) $(HOST_LDFLAGS) -o $@ $^ -lprotobuf-c
 
-$(OUTPUT)/liberpal-skeleton-v2.o: liberpal-skeleton-v2.c liberpal-skeleton.c
-	$(CC) $(HOST_CFLAGS) -c $< -o $@
-
-$(OUTPUT)/liberpal-skeleton.o: liberpal-skeleton.c
-	$(CC) $(HOST_CFLAGS) -c $< -o $@
+$(OUTPUT)/liberpal-skeleton-v3.so: $(OUTPUT)/sgx_call.o $(OUTPUT)/liberpal-skeleton-v3.o $(OUTPUT)/liberpal-skeleton.o $(OUTPUT)/sgxutils.o $(OUTPUT)/aesm.o $(OUTPUT)/aesm.pb-c.o
+	$(CC) $(HOST_LDFLAGS) -o $@ $^ -lprotobuf-c
 
-$(OUTPUT)/liberpal-skeleton-v3.so: $(OUTPUT)/sgx_call.o $(OUTPUT)/liberpal-skeleton-v3.o $(OUTPUT)/liberpal-skeleton.o $(OUTPUT)/sgxutils.o
-	$(CC) $(HOST_LDFLAGS) -o $@ $^
+$(OUTPUT)/$(OBJS): %.o: %.c
+	$(CC) $(HOST_CFLAGS) -c $< -o $@ -lprotobuf-c
 
-$(OUTPUT)/liberpal-skeleton-v3.o: liberpal-skeleton-v3.c liberpal-skeleton.c
-	$(CC) $(HOST_CFLAGS) -c $< -o $@
+$(OUTPUT)/aesm.pb-c.c: aesm.proto
```

Comment:
aesm.pb-c.c should be recorded in .gitignore because it is an intermediate file.